ArduCopter/landing_gear.cpp

```diff
@@ -18,6 +18,21 @@ void Copter::landinggear_update()
         landinggear.set_position(AP_LandingGear::LandingGear_Deploy_And_Keep_Deployed);
     }
 
+    // if landing gear is locked down from an automatic deployment, and we're
+    // no longer in mode that requires it, check the status of the gear switch
+    // and release the lock if the switch isn't still up.
+    if (!flightmode->landing_gear_should_be_deployed() && landinggear.locked()) {
+        if (gear_sw_up) {
+            if (!gcs_gear_sw_warned) {
+                gcs().send_text(MAV_SEVERITY_WARNING, "Check Landing Gear Switch");
```

Comment:
mavlink is a lossy protocol.  This should probably be emitted every minute or so.  Not a huge thing, 'though.